### PR TITLE
Updated picnic to v2.2.

### PR DESCRIPTION
**Pull request by christianpaquin**
Updated picnic to v2.2. From the changelog:

- Fix Picnic2 implementation on big endian systems
- Add support for SHA3/SHAKE3 instructions on IBM z.
- Various small improvements and bug fixes.
- Remove LowMC instances with m=1.

### CODE REVIEW

**Comment by christianpaquin** [comment]
Pausing review for now; checking to see if more is needed to fix the OQS/OpenSSL picnic issues.

**Comment by christianpaquin** [comment]
Looks like clang-9 works with unmodified v2.2. @xvzcf, can you confirm?

**Comment by christianpaquin** [comment]
The Picnic team just notified me they backported a clang-9 bug fix from v3 to v2.2, so I'll include that commit as well...

**Comment by christianpaquin** [comment]
Ok, should be finally ready for review @xvzcf, @dstebila. That should fix the picnic issue.

File: src/sig/picnic/CMakeLists.txt [hunk 1]
```diff
@@ -81,9 +70,6 @@ target_compile_definitions(picnic PRIVATE PICNIC_STATIC
                                           WITH_LOWMC_192_192_30
                                           WITH_LOWMC_256_256_38
                                           WITH_OPT)
-if(NOT WIN32)
-    target_compile_definitions(picnic PRIVATE HAVE_POSIX_MEMALIGN)
-endif()
 
 if(OQS_USE_SSE2_INSTRUCTIONS AND USE_OPTIMIZATIONS)
```
Comment (xvzcf) [inline src/sig/picnic/CMakeLists.txt:74]: You should remove all references to `USE_OPTIMIZATIONS` since it's no longer set anywhere.

**Comment by xvzcf** [comment]
All tests pass on `clang-10` with the requested changes made.

File: src/sig/picnic/CMakeLists.txt (hunk 1 above)
Comment (christianpaquin) [reply src/sig/picnic/CMakeLists.txt]: Done

**Comment by xvzcf** [comment]
Looks good for me. I'll wait for @dstebila's approval.

**Pull request merged**